### DISCUSSION TITLE
Haxophone software with transpose

### DISCUSSION THREAD

**Discussion by ellfyy** [cardonabits/haxo-hw#83] [Q&A]
Hi, I would like to use the tranpose feature but am completely horrendous at software side and dont understand the way to compile the software myself. Is there a premade software with transpose available that I can use with rpizero2 please? Or an easier way for me to add transpose to my software? Thanks :)

**Comment by jcard0na** [comment]
Hi @ellfyy,

The Pi Zero binaries we build on github should have the latest features and have been confirmed to work on Pi Zero 2.  Can you try to install that on your image?  See https://github.com/cardonabits/haxo-hw/discussions/74
You will also need to copy over `notemap.json` from [here](https://github.com/cardonabits/haxo-rs/blob/main/notemap.json) onto your haxophone, under `/usr/share/haxo`, which was changed by this commit https://github.com/cardonabits/haxo-rs/pull/27

Cheers,

**Reply by ellfyy** [reply]
Okay will try this now and update once I finish. Thanks :)

**Reply by ellfyy** [reply]
Hi so sorry again, how do I do the replace binary part. I cant seem to work it out sorry. :)

**Reply by stardusthamster** [reply]
If you wait a bit, I am currently writing a manual for the haxophone and I will include this part.

For a sneak preview go to https://github.com/stardusthamster/haxo-hw/blob/2-manual-continued/docs/manual/manual.md

In the current version the explanation for update of the binary is still very short and I will explain it in further detail.

**Reply by StaffanMelin** [reply]
Great work! @jcard0na, I suggest enabling the WIKI on the haxo-rs/hw repos and maybe put @stardusthamster's manual there? If you want to @stardusthamster ?

**Reply by stardusthamster** [reply]
The manual is updated!

As for the wiki option: For my part I would like to let @jcard0na decide.

With the manual I want to contribute a bit to this great project!

**Reply by StaffanMelin** [reply]
@stardusthamster: what is the URL?

**Reply by jcard0na** [reply]
https://github.com/cardonabits/haxo-hw/blob/main/docs/manual/manual.md

**Comment by stardusthamster** [comment]
@StaffanMelin: Have a look at the manual and tell me if there is something missing or how you did get along. Feedback is appreciated!

**Comment by StaffanMelin** [comment]
Looks great @stardusthamster! I really appreciate having all bits and pieces collected like this. And the new photos are very helpful too. That cheat sheet is great! https://github.com/cardonabits/haxo-hw/blob/main/docs/manual/cheatsheet.pdf

Just off the top of my head:

When you mention playing/adding a tenor sax mouthpiece, maybe stress that it should have a reed, else it will leak and not work.

The latency introduced by connecting over WIFI via SSH is not important, as no audio is transferred. This is only for hacking it. I prefer WIFI as I have attached a RPi 4B, with builtin WIFI. Enabling it is easy. And you will not need any anoying dongle hanging from your sax/hax.

Add info on how to position the tube hanging down so you wont short circuit your hax. https://github.com/cardonabits/haxo-hw/issues/80

And as soon as my "alternative firmware" matures, maybe you could add that! :D (Thank you for the star :) ). (https://github.com/StaffanMelin/haxo-cs). I have some interesting ideas on how to use the hax as a "not-a-sax"-controller too, and have started some work on my expression pedal for the hax (with analog outs so it can be used with modular analog gear).

**Reply by stardusthamster** [reply]
Thank you for your feedback! 
I will include it in the next version. But it will take me some time ...
And I am looking forward to your "alternative firmware"!
